### ISSUE TITLE
Visual Studio for Mac: LocalDB is not supported on this platform.

### ISSUE THREAD

**Issue by kendrickhang** [dotnet/AspNetCore.Docs#16561]
When my colleagues and I run this tutorial on our Macs (latest VS 2019 for Mac, Community Edition; .NET 3.1.1), we get an "LocalDB is not supported on this platform." error when running the "dotnet ef database update" step on the command line.

When I ran the tutorial a few months ago, I noticed the generated code in Startup.cs was referring to Sqlite:
services.AddDbContext&lt;RazorPagesMovieContext&gt;(options =&gt;
                options.UseSqlite(Configuration...

This go around, I'm noticing it's referring to SqlServer:
services.AddDbContext&lt;RazorPagesMovieContext&gt;(options =&gt;
                options.UseSqlServer(Configuration...

Is there a step we can take on the macs to use SqlLite in the tutorial to have it work?

Appreciate any guidance/feedback/help.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 6719f08e-3bd7-dc1a-71df-f2ef9fbca9d8
* Version Independent ID: 7096fdb3-612e-9e00-bd0b-8ea4886a09ce
* Content: [Add a model to a Razor Pages app in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/tutorials/razor-pages/model?view=aspnetcore-3.1&tabs=visual-studio-mac#feedback)
* Content Source: [aspnetcore/tutorials/razor-pages/model.md](https://github.com/aspnet/AspNetCore.Docs/blob/master/aspnetcore/tutorials/razor-pages/model.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-tutorials**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

**Comment by Rick-Anderson** [comment]
Follow the Mac instructions

**Comment by kendrickhang** [comment]
Hi Rick, yes, I did, and I had three other developers do it with me. There is something that is not matching up. Thanks for taking a look if you can. -Ken

**Comment by Rick-Anderson** [comment]
Our high school interns went through this using mac's. You need to select the Mac tab. We have lots of mac customers reporting success.

**Comment by talmadge-simpson** [comment]
I too am running into the same issue. I when through the process twice with the same result and verified that I am following the MAC tab. I see that this issue is closed. How was the matter resolve?
Please advise
Thank you

**Comment by Rick-Anderson** [comment]
@sayedihashimi  customers are reporting your latest changes are not working.

**Comment by Rick-Anderson** [comment]
Moved to Master Issue #15941

**Comment by cmh0098** [comment]
Was this issue ever resolved for Mac?

**Comment by azuc-ct** [comment]
Hey folks, I started jumping around the different steps of this tutorial while stuck on this issue as well and may have found something that worked that's already documented in the tutorial:a

+ When you are on “Add a Model” and are specifically in the “Scaffold the movie model” and “Initial migration” sections, the scaffolding will build a connection string referencing (localdb)\\mssqllocaldb. If you use what the scaffolding automatically generates, you will get a “LocalDB is not supported on this platform.” error when you try to update with your migration (dotnet ef database update). If that happens, move on:

+ In that tutorial, jump ahead to “Work with a database” and switch the tab from “Visual Studio” to “Visual Studio Code/Visual Studio for Mac” and you will see that there are changes to make to Startup.cs and appsettings.json regarding your connection string and type of SQL connection. Make those changes - you will likely see a compilation error with the UseSqlite method. If that happens, do this last thing:

+ Hop in to your nuget directory under the Dependencies directory of your RazorPagesMovies solution and go to Manage Nuget Packages. Add the package “Microsoft.EntityFrameworkCore.Sqlite.” This basically adds the following to one of the <ItemGroup> fields in your csproj: 
<PackageReference Include="Microsoft.EntityFrameworkCore.Sqlite" Version="3.1.1" />
You should see your UseSqlite method compile fine in Startup.cs.

+ Now rerun dotnet ef database update and fire up your application as you were instructed in the “Add a model” section and you should be all good to go.

**Comment by talmadge-simpson** [comment]
I have not revisited the Razor Pages tutorial. I was able to overcome the issue by following the MVC tutorial. I cannot in good conscious clear you to close the issue. In the terms of software development, this was so long ago and I have advanced so far in my project that I can not return to verify any corrections you or your team may have made. I am sure you understand.

 

While I have your attention, I would like to thank you and Microsoft for providing these tutorials. I have found them to be instrumental in overcoming the learning curve associated with expanding my knowledge of ASP.NET Core and practical skillset.

 

Thank you for following up

 

August Talmadge Simpson

talmadgesimpson.com

(551) 208-5909

2453 Tropical Shores Dr SE

St Petersburg, FL 33705
